test/tests/functional/pbs_mom_walltime.py

```diff
@@ -147,17 +147,26 @@ def test_suspend_time_not_counted_in_walltime(self):
         self.server.manager(MGR_CMD_SET, MGR_OBJ_SERVER,
                             {'scheduling': 'True'})
 
-        time.sleep(15)
+        time.sleep(30)
 
         # Used walltime should remain the same
         self.server.expect(JOB, {'resources_used.walltime': walltime}, op=EQ,
-                           id=jid1, extend='x')
+                           id=jid1)
 
         self.server.sigjob(jobid=jid1, signal="resume")
         self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
         time.sleep(10)
         self.server.expect(JOB, {'resources_used.walltime': walltime}, op=GT,
                            id=jid1, extend='x')
+        self.server.expect(JOB, {ATTR_state: 'F'}, id=jid1, extend='x',
+                           offset=20)
+
+        # Verify if the job's total walltime is within limits
+        # Adding 10s buffer since min mom poll time is 10s
+        self.server.expect(JOB, {'resources_used.walltime': 25}, op=GE,
+                           id=jid1, extend='x')
```

Comment:
Since the walltime shoulnd't update while the job is in the F state, you should put max_attempts=1 for both of these expects.